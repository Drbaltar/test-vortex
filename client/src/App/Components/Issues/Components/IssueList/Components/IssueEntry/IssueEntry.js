import React from 'react';
import ListEntry from '../../../../../shared-components/ListEntry/ListEntry';

const IssueEntry = (props) => {
    const { entry, index, select } = props;

    return <ListEntry entryFields={getIssueFields(entry)} index={index} select={select}/>
};

const getIssueFields = (issue) => {
    return [
        {
            label: 'Issue Type:',
            field: issue.issue_type,
        },
        {
            label: 'Issue Category:',
            field: issue.issue_category
        },
        {
            label: 'Issue Description:',
            field: issue.issue_description
        },
        {
            label: 'Submitted By:',
            field: issue.submitted_by
        }
    ];
}; 

export default IssueEntry;
