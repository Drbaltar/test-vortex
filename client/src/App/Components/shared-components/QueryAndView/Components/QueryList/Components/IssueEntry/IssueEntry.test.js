import React from 'react';
import { shallow } from 'enzyme';
import IssueEntry from './IssueEntry';

describe('IssueEntry', () => {
    const testEntry = {
        issue_type: 'Application Issue',
        issue_category: 'Application Crash',
        issue_description: 'The application keeps crashing!',
        submitted_by: 'kmccain'
    };
    const testIndex = 5;
    const mockSelect = jest.fn();

    const wrapper = shallow(<IssueEntry entry={testEntry} select={mockSelect} index={testIndex}/>);
    const listEntryFields = wrapper.find('ListEntry').prop('entryFields');

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('passes the select function to ListEntry', () => {
        expect(wrapper.find('ListEntry').prop('select')).toBe(mockSelect);
    });

    it('passes the index to ListEntry', () => {
        expect(wrapper.find('ListEntry').prop('index')).toEqual(testIndex);
    });

    it('passes the proper number of fields to ListEntry', () => {
        expect(listEntryFields.length).toBe(4);
    });

    it('passes the Issue Type label and field correctly', () => {
        expect(listEntryFields[0].label).toBe('Issue Type:');
        expect(listEntryFields[0].field).toBe(testEntry.issue_type);
    });

    it('passes the Issue Category label and field correctly', () => {
        expect(listEntryFields[1].label).toBe('Issue Category:');
        expect(listEntryFields[1].field).toBe(testEntry.issue_category);
    });

    it('passes the Issue Description label and field correctly', () => {
        expect(listEntryFields[2].label).toBe('Issue Description:');
        expect(listEntryFields[2].field).toBe(testEntry.issue_description);
    });

    it('passes the Submitted By label and field correctly', () => {
        expect(listEntryFields[3].label).toBe('Submitted By:');
        expect(listEntryFields[3].field).toBe(testEntry.submitted_by);
    });

    describe('when the select function is called in the list entry component', () => {
        beforeAll(() => {
            const select = wrapper.find('ListEntry').prop('select');
            select(testIndex);
        });

        it('the select function should be called with the index', () => {
            expect(mockSelect).toHaveBeenCalledWith(testIndex);
        });
    });
});