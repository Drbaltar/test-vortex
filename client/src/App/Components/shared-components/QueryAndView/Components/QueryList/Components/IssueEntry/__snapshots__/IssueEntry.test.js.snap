// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueEntry renders correctly 1`] = `
<ListEntry
  entryFields={
    Array [
      Object {
        "field": "Application Issue",
        "label": "Issue Type:",
      },
      Object {
        "field": "Application Crash",
        "label": "Issue Category:",
      },
      Object {
        "field": "The application keeps crashing!",
        "label": "Issue Description:",
      },
      Object {
        "field": "kmccain",
        "label": "Submitted By:",
      },
    ]
  }
  index={5}
  select={[MockFunction]}
/>
`;
